src/service/metrics/otlp.rs

```diff
@@ -919,3 +950,42 @@ fn process_aggregation_temporality(rec: &mut json::Value, val: i32) {
     }
     .into();
 }
+
+fn format_response(
+    mut partial_success: ExportMetricsPartialSuccess,
+    req_type: OtlpRequestType,
+) -> Result<HttpResponse, anyhow::Error> {
+    let partial = partial_success.rejected_data_points > 0;
+
+    let res = if partial {
+        log::error!(
+            "[METRICS:OTLP] Partial success: {}, error: {}",
+            partial_success.rejected_data_points,
+            partial_success.error_message
+        );
+        partial_success.error_message =
+            "Some data points were rejected due to exceeding the allowed retention period"
+                .to_string();
```

Comment:
I don't  think we should overwrite the error message here, in two cases above where there is partial response, it might not necessarily be due to retention period, and we are setting appropriate error in the partial success struct.

Reply:
yes, but we will be replying with this error instead of the actual to the user right? Shouldn't we send the actual error in the response as well?

Reply:
Maybe, but i copy this from traces, i think for some special reason we set it like this. but i didn't recall, so let's keep it for a while.